sprints/dashboard/automation.py

```diff
@@ -0,0 +1,257 @@
+
+    The injection is determined by checking the last update of the "Sprint" value in the ticket.
+    Therefore, if the ticket has been moved out of the sprint for a moment after the cutoff date, and then added back,
+    it is still considered as an injection.
+
+    The cutoff date is configured via the `SPRINT_ASYNC_TICKET_CREATION_CUTOFF_DAY` env variable.
+    The injection can be "accepted" (ignored) by adding a `SPRINT_ASYNC_INJECTION_LABEL` to the ticket.
+
+    Note: the link below and API docs indicate that the `changelog` contains all existing changes - the pagination has
+    `maxResults = total` by default. This was tested this with a ticket that has over 3500 items in the changelog and it
+    worked as described. Most of these tickets should be fairly new, so they will not have such extensive
+    changelogs, but it is worth to keep this in mind, as not all behaviors of the API are documented properly.
+    https://community.atlassian.com/t5/Answers-Developer-Questions/How-to-get-complete-changelog-for-a-issue/qaq-p/501636]
+
+    :param conn: Jira connection.
+    :param issue: Jira issue.
+    :return: Is the ticket an injection?
+    """
+    # The injection can to be "accepted" with a label.
+    if settings.SPRINT_ASYNC_INJECTION_LABEL in getattr(issue.fields, conn.issue_fields[settings.JIRA_FIELDS_LABELS]):
+        return False
+
+    cutoff_date = get_specific_day_of_sprint(settings.SPRINT_ASYNC_TICKET_CREATION_CUTOFF_DAY).strftime(
+        "%Y-%m-%dT%H:%M:%S"
+    )
+
+    for history in reversed(issue.changelog.histories):
+        # The updates are ordered from the newest to the oldest, so we can skip the ones done before the cutoff date.
+        if history.created < cutoff_date:
+            break
+
+        if history.items[0].field == "Sprint":
+            return True
+
+    return False
+
+
+def check_issue_missing_fields(conn: CustomJira, issue: Issue) -> list[str]:
+    """
+    Get a list of fields that are missing from a ticket to call it ready for a sprint.
+
+    :param conn: Jira connection.
+    :param issue: Jira issue.
+    :return: List of missing fields.
+    """
+    messages = []
+    # TODO: Make this configurable. It might need some tweaks to handle default behaviors for various fields.
+    for field in [settings.JIRA_FIELDS_ASSIGNEE, settings.JIRA_FIELDS_REVIEWER, settings.JIRA_FIELDS_STORY_POINTS]:
+        if not getattr(issue.fields, conn.issue_fields[field]):
+            messages.append(field)
+
+    return messages
+
+
+def get_overcommitted_users(conn: CustomJira) -> dict[str, list[User]]:
+    """
+    Retrieve users that have negative time left for the next sprint (i.e. are overcommitted).
+
+    :param conn: Jira connection.
+    :return: List of overcommitted users.
+    """
+    cells = get_cells(conn)
+    result = dict[str, list[User]]()
+
+    # Generate dashboards in parallel.
+    with ThreadPool(processes=settings.MULTIPROCESSING_POOL_SIZE) as pool:
```

Comment:
It doesn't look that elegant, but for 3 cells it optimizes this function from 50s to 20s